nupic/research/frameworks/continuous_learning/utils.py

```diff
@@ -108,59 +115,22 @@ def train_model(
     loader,
     optimizer,
     device,
-    freeze_params=None,
-    freeze_fun=None,
-    freeze_pct=90,
+    sample_fraction=None,
+    normalize_input=False,
+    freeze_modules=None,
+    module_inds=None,
     freeze_output=False,
     layer_type="dense",
     linear_number=2,
     output_indices=None,
-    duty_cycles=None,
-    frozen_dcs=None,
-    reset_weights=None,
-    reset_fun=None,
-    reset_params=None,
     criterion=F.nll_loss,
     batches_in_epoch=sys.maxsize,
     pre_batch_callback=None,
     post_batch_callback=None,
     progress_bar=None,
     combine_data=False,
 ):
-    """Train the given model by iterating through mini batches. An epoch ends
-    after one pass through the training set, or if the number of mini batches
-    exceeds the parameter "batches_in_epoch".
-
-    :param model: pytorch model to be trained
-    :type model: torch.nn.Module
-    :param loader: train dataset loader
-    :type loader: :class:`torch.utils.data.DataLoader`
-    :param optimizer: Optimizer object used to train the model.
-           This function will train the model on every batch using this optimizer
-           and the :func:`torch.nn.functional.nll_loss` function
-    :param batches_in_epoch: Max number of mini batches to train.
-    :param device: device to use ('cpu' or 'cuda')
-    :type device: :class:`torch.device
-    :param freeze_params: List of parameters to freeze at specified indices
-     For each parameter in the list:
-     - parameter[0] -> network module
-     - parameter[1] -> weight indices
-    :type param: list or tuple
-    :param criterion: loss function to use
-    :type criterion: function
-    :param post_batch_callback: Callback function to be called after every batch
-                                with the following parameters: model, batch_idx
-    :type post_batch_callback: function
-    :param pre_batch_callback: Callback function to be called before every batch
-                               with the following parameters: model, batch_idx
-    :type pre_batch_callback: function
-    :param progress_bar: Optional :class:`tqdm` progress bar args.
-                         None for no progress bar
-    :type progress_bar: dict or None
-
-    :return: mean loss for epoch
-    :rtype: float
-    """
+
```

Comment:
Why remove the docstring? At the very least the new parameters should be documented. Not clear to me at all what they mean.